apps/hash-graph/lib/graph/src/snapshot.rs

```diff
@@ -113,10 +115,12 @@ where
     )]
     pub async fn restore_snapshot(
         &mut self,
-        snapshot: impl Stream<Item = SnapshotEntry> + Send,
+        snapshot: impl Stream<Item = Result<SnapshotEntry, impl Context>> + Send,
```

Comment:
This was the easiest possibility I've seen to notify the stream, that it's canceled.

Reply:
I assume you mean notifying the consumer of the stream that the stream is about to be canceled? (-> having a fallible stream)

Reply:
Yes, exactly. the stream could be abortable, but this requires the handle to be passed around.

Reply:
In that case, how about using a [`TryStream`](https://docs.rs/futures/latest/futures/prelude/stream/trait.TryStreamExt.html)?

Reply:
If you do not need to pass any information, there are also [`CancellationToken`](https://docs.rs/tokio-util/latest/tokio_util/sync/struct.CancellationToken.html)s

Reply:
This is a `TryStream` 😄 
```rust
impl<S, T, E> TryStream for S
where
    S: Stream<Item = Result<T, E>> + ?Sized;
```

Reply:
Ay, but I specifically meant the `try_next` method from `TryStreamExt`.

Reply:
Yes, but this is exactly what `TryStream`s are, a `Stream` with a `Result`. `TryStreamExt` only provides some convenient functions on `Ok` or `Err` variants.

> If you do not need to pass any information, there are also [CancellationToken](https://docs.rs/tokio-util/latest/tokio_util/sync/struct.CancellationToken.html)s

I was thinking about [`AbortHandle`](https://docs.rs/futures/latest/futures/stream/struct.AbortHandle.html), but I think `CancellationToken` should also work. I will take a look at these, thanks! (but not in this PR)

Reply:
With `TryStream` you need to pass in some information but sometimes you don't even have an error type.